File: src/main/java/priv/bajdcc/OP/syntax/precedence/FixedData.java
package priv.bajdcc.OP.syntax.precedence;

import priv.bajdcc.util.lexer.token.Token;

/**
 * 混合数据结构
 *
 * @author bajdcc
 */
public class FixedData {

	/**
	 * 单词
	 */
	public Token token = null;

	/**
	 * 数据
	 */
	public Object obj = null;

	public FixedData() {

	}

	public FixedData(Token token) {
		this.token = token;
	}

	public FixedData(Object obj) {
		this.obj = obj;
	}

	@Override
	public String toString() {
		if (token != null) {
			return token.toString();
		} else if (obj != null) {
			return obj.toString();
		} else {
			return "(null)";
		}
	}
}

File: src/main/kotlin/com/bajdcc/OP/syntax/precedence/FixedData.kt
package com.bajdcc.OP.syntax.precedence

import com.bajdcc.util.lexer.token.Token

/**
 * 混合数据结构
 *
 * @author bajdcc
 */
class FixedData {

    /**
     * 单词
     */
    var token: Token? = null

    /**
     * 数据
     */
    var obj: Any? = null

    constructor()

    constructor(token: Token) {
        this.token = token
    }

    constructor(obj: Any?) {
        this.obj = obj
    }

    override fun toString(): String {
        return if (token != null) {
            token!!.toString()
        } else if (obj != null) {
            obj!!.toString()
        } else {
            "(null)"
        }
    }
}
